Sources/Private/Experimental/Animations/ShapeAnimation.swift

```diff
@@ -1,28 +1,81 @@
-// Created by Cal Stephens on 12/21/21.
-// Copyright © 2021 Airbnb Inc. All rights reserved.
+// Created by Cal Stephens on 1/7/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
 
 import QuartzCore
 
-extension CALayer {
-  /// Adds animations for the given `Shape` to this `CALayer`
-  func addAnimations(
-    for shape: Shape,
-    context: LayerAnimationContext)
-  {
-    addAnimations(for: shape.path, context: context)
+extension CAShapeLayer {
+  /// Adds a `path` animation for the given `ShapeItem`
+  func addAnimations(for shape: ShapeItem, context: LayerAnimationContext) {
+    switch shape {
+    case let customShape as Shape:
+      addAnimations(for: customShape.path, context: context)
+
+    case let ellipse as Ellipse:
+      addAnimations(for: ellipse, context: context)
+
+    case let rectangle as Rectangle:
+      addAnimations(for: rectangle, context: context)
+
+    case let star as Star:
+      addAnimations(for: star, context: context)
+
+    default:
+      return
+    }
+  }
+
+  /// Adds a `fillColor` animation for the given `Fill` object
+  func addAnimations(for fill: Fill, context: LayerAnimationContext) {
```

Comment:
Moved these `ShapeItem`-related animations to `ShapeAnimation.swift`, from `ShapeItemLayer.swift`